contrib/src/main/java/com/datatorrent/contrib/hive/FSRollingPOJOImplementation.java

```
@@ -0,0 +1,248 @@
+  private ArrayList<String> hiveColumns;
+  private ArrayList<FIELD_TYPE> hiveColumnsDataTypes;
+  private transient ArrayList<Object> getters;
+  private ArrayList<String> expression;
+
+  /*
+   * A list of Java expressions in which each expression yields the specific table column value and partition column value in hive table from the input POJO.
+   */
+  public ArrayList<String> getExpression()
+  {
+    return expression;
+  }
+
+  public void setExpression(ArrayList<String> expression)
+  {
+    this.expression = expression;
+  }
+
+  public FSRollingPOJOImplementation()
+  {
+    super();
+    getters = new ArrayList<Object>();
+  }
+
+  private Object getGetter(Object tuple, int index, FIELD_TYPE type)
+  {
+    Object getter;
+    switch (type) {
+      case CHARACTER:
+        getter = ((GetterChar)getters.get(index)).get(tuple);
+        break;
+      case STRING:
+        getter = ((GetterString)getters.get(index)).get(tuple);
+        break;
+      case BOOLEAN:
+        getter = ((GetterBoolean)getters.get(index)).get(tuple);
+        break;
+      case SHORT:
+        getter = ((GetterShort)getters.get(index)).get(tuple);
+        break;
+      case INTEGER:
+        getter = ((GetterInt)getters.get(index)).get(tuple);
+        break;
+      case LONG:
+        getter = ((GetterLong)getters.get(index)).get(tuple);
+        break;
+      case FLOAT:
+        getter = ((GetterFloat)getters.get(index)).get(tuple);
+        break;
+      case DOUBLE:
+        getter = ((GetterDouble)getters.get(index)).get(tuple);
+        break;
+      case DATE:
+        getter = (Date)((GetterObject)getters.get(index)).get(tuple);
+        break;
+      case TIMESTAMP:
+        getter = (Timestamp)((GetterObject)getters.get(index)).get(tuple);
+        break;
+      case OTHER:
+        getter = (Timestamp)((GetterObject)getters.get(index)).get(tuple);
+        break;
+      default:
+        getter = ((GetterObject)getters.get(index)).get(tuple);
```

What is the difference between OTHER and default? If you handle all known/supported types, default should throw an exception.